clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -424,6 +424,14 @@ public void testPercentiles() {
         assertEquals(0.0, p25.value(), 1.0);
         assertEquals(0.0, p50.value(), 1.0);
         assertEquals(0.0, p75.value(), 1.0);
+
+        // record two more windows worth of sequential values
```

Comment:
> Do you know why the existing test didn't trigger this issue when we added 2 more windows of 0?

When histogram is not reset, incorrect histogram bucket is selected for a given quantile: simply put, as the samples are purged, instead of p50 you get (approximate) value for p25, then p12.5 and so on. But when two windows of zeros are recorded, it does not matter, since all percentiles have the same value: zero.

The test with zeroes looks useful though, as it clearly shows that the percentiles depend on the recent data only.